ecl/hql/hqlutil.cpp

```diff
@@ -8855,12 +8859,14 @@ bool ConstantRowCreator::processFieldValue(IHqlExpression * optLhs, ITypeInfo *
     case type_groupedtable:
         {
             assertex(optLhs);
+            IHqlExpression * defaultAttrib = optLhs->queryAttribute(defaultAtom);
             IHqlExpression * field = optLhs->queryChild(1);
-            if (!field->hasAttribute(countAtom) && !field->hasAttribute(sizeofAtom))
+            if (defaultAttrib || (!field->hasAttribute(countAtom) && !field->hasAttribute(sizeofAtom)))
             {
-                if (field->hasAttribute(_linkCounted_Atom))
+                // There may be no field when default value is provided
+                if (field && field->hasAttribute(_linkCounted_Atom))
                 {
-                    if (rhsOp == no_null)
+                    if (rhsOp == no_null || (defaultAttrib && isNullList(defaultAttrib->queryChild(0))))
```

Comment:
This code looks out of place - it is likely to build null datasets in the wrong places.